core/src/main/java/eu/fasten/core/maven/resolution/MavenDependencyResolver.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2022 Delft University of Technology
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package eu.fasten.core.maven.resolution;
+
+import static eu.fasten.core.maven.data.Scope.COMPILE;
+import static eu.fasten.core.maven.data.Scope.PROVIDED;
+import static eu.fasten.core.maven.data.Scope.RUNTIME;
+import static eu.fasten.core.maven.data.Scope.SYSTEM;
+import static eu.fasten.core.maven.data.Scope.TEST;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import eu.fasten.core.maven.data.Dependency;
+import eu.fasten.core.maven.data.MavenProduct;
+import eu.fasten.core.maven.data.Pom;
+import eu.fasten.core.maven.data.ResolvedRevision;
+import eu.fasten.core.maven.data.Scope;
+import eu.fasten.core.maven.data.VersionConstraint;
+
+public class MavenDependencyResolver {
+
+    private MavenDependencyData graph = new MavenDependencyData();
+
+    public void setData(MavenDependencyData graph) {
+        this.graph = graph;
+    }
+
+    public Set<ResolvedRevision> resolve(Collection<String> gavs, ResolverConfig config) {
+
+        failForImportProvidedSystemScope(config);
+
+        if (gavs.size() == 1) {
+            var parts = gavs.iterator().next().split(":");
+            var ga = String.format("%s:%s", parts[0], parts[1]);
+
+            var pom = this.graph.find(ga, Set.of(VersionConstraint.init(parts[2])), config.resolveAt);
+            if (pom == null) {
+                throw new MavenResolutionException(
+                        String.format("Cannot find coordinate %s:%s:%s", parts[0], parts[1], parts[2]));
+            } else {
+                return resolve(config, new HashSet<>(), QueueData.startFrom(pom));
```

Comment:
The variable is just used for preventing an endless loop. I do not think that an assignment would make it easier to read.

Reply:
Yeah, that's true, it may not make it easier to read. But I usually try to lower the number of arguments whenever possible as a good practice, I think it is more because of maintainability and usability of the method rather than readability.

Reply:
No, it is not possible to leave it out, as it is handed down in the recursive calls.